internal/controller/atlasnetworkpeering/atlasnetworkpeering_controller.go

```diff
@@ -0,0 +1,433 @@
+	}
+
+	if !r.AtlasProvider.IsResourceSupported(&akoNetworkPeering) {
+		return r.Unsupport(workflowCtx, typeName)
+	}
+
+	credentials, err := r.ResolveCredentials(ctx, &akoNetworkPeering)
+	if err != nil {
+		return r.terminate(workflowCtx, &akoNetworkPeering, workflow.AtlasAPIAccessNotConfigured, err), nil
+	}
+	sdkClient, orgID, err := r.AtlasProvider.SdkClient(ctx, credentials, r.Log)
+	if err != nil {
+		return r.terminate(workflowCtx, &akoNetworkPeering, workflow.AtlasAPIAccessNotConfigured, err), nil
+	}
+	project, err := r.ResolveProject(ctx, sdkClient, &akoNetworkPeering, orgID)
+	if err != nil {
+		return r.terminate(workflowCtx, &akoNetworkPeering, workflow.AtlasAPIAccessNotConfigured, err), nil
+	}
+	return r.handle(&reconcileRequest{
+		workflowCtx:    workflowCtx,
+		service:        networkpeering.NewNetworkPeeringService(sdkClient.NetworkPeeringApi),
+		projectID:      project.ID,
+		networkPeering: &akoNetworkPeering,
+	}), nil
+}
+
+func (r *AtlasNetworkPeeringReconciler) handle(req *reconcileRequest) ctrl.Result {
+	r.Log.Infow("handling network peering reconcile request",
+		"service set", (req.service != nil), "projectID", req.projectID, "networkPeering", req.networkPeering)
+	var atlasPeer *networkpeering.NetworkPeer
+	if req.networkPeering.Status.ID != "" {
+		peer, err := req.service.GetPeer(req.workflowCtx.Context, req.projectID, req.networkPeering.Status.ID)
+		if err != nil && !errors.Is(err, networkpeering.ErrNotFound) {
+			return r.terminate(req.workflowCtx, req.networkPeering, workflow.Internal, err)
+		}
+		atlasPeer = peer
+	}
+	inAtlas := atlasPeer != nil
+	deleted := req.networkPeering.DeletionTimestamp != nil
+
+	switch {
+	case !deleted && !inAtlas:
+		return r.create(req)
+	case !deleted && inAtlas:
+		return r.sync(req, atlasPeer)
+	case deleted && inAtlas:
+		return r.delete(req, atlasPeer)
+	default:
+		return r.unmanage(req) // deleted && !inAtlas
+	}
+}
+
+func (r *AtlasNetworkPeeringReconciler) create(req *reconcileRequest) ctrl.Result {
+	container, err := r.handleContainer(req)
+	if err != nil {
+		return r.terminate(req.workflowCtx, req.networkPeering, workflow.Internal, err)
+	}
+	req.workflowCtx.EnsureStatusOption(updateContainerStatusOption(container))
+	specPeer := networkpeering.NewNetworkPeeringSpec(&req.networkPeering.Spec.AtlasNetworkPeeringConfig)
+	specPeer.ContainerID = container.ID
+	newPeer, err := req.service.CreatePeer(req.workflowCtx.Context, req.projectID, specPeer)
+	if err != nil {
+		return r.terminate(req.workflowCtx, req.networkPeering, workflow.Internal, err)
+	}
+	req.workflowCtx.EnsureStatusOption(updatePeeringStatusOption(newPeer))
+	return workflow.InProgress(
+		workflow.NetworkPeeringConnectionCreating,
+		fmt.Sprintf("Network Peering Connection %s is %s",
+			req.networkPeering.Status.ID, req.networkPeering.Status.Status),
+	).ReconcileResult()
+}
+
+func (r *AtlasNetworkPeeringReconciler) sync(req *reconcileRequest, atlasPeer *networkpeering.NetworkPeer) ctrl.Result {
```

Comment:
blocking: what is being `sync`'ed here? as I read the code, updates in the spec are silently ignored. If the spec is immutable, we should talk w/ PM whether it makes sense to make the whole spec immutable (guarded via CEL) if Atlas doesn't support updates to network peers.

Reply:
I see we have updates in Atlas for both connections https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Network-Peering/operation/updatePeeringConnection and containers https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Network-Peering/operation/updatePeeringContainer.

Why aren't we leveraging that?